File: VB/ViewModel/MainViewModel.vb
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.POCO

Namespace Example.ViewModel

    <POCOViewModel>
    Public Class MainViewModel

        Private SavedText As String

        Public Overridable Property Text As String

        Public ReadOnly Property IsSaved As Boolean
            Get
                Return Equals(Text, SavedText)
            End Get
        End Property

        Public Sub Save()
            SavedText = Text
            RaisePropertyChanged(Function(x) x.IsSaved)
        End Sub

        Public Sub Close()
            SavedText = ""
            Text = ""
        End Sub

        Protected Sub OnTextChanged()
            RaisePropertyChanged(Function(x) x.IsSaved)
        End Sub
    End Class
End Namespace

File: CS/ViewModel/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;

namespace Example.ViewModel
{
    public class MainViewModel : ViewModelBase {
        
        string savedText;
        
        public string Text {
            get => GetValue<string>();
            set {
                SetValue(value);
                RaisePropertyChanged(nameof(IsSaved));
            }
        }
        public bool IsSaved {
            get {
                return Text == savedText && !string.IsNullOrEmpty(Text);
            }
        }

        [Command]
        public void Save() {
            savedText = Text;
            RaisePropertyChanged(nameof(IsSaved));
        }
        
        [Command]
        public void Close() {
            savedText = string.Empty;
            Text = string.Empty;
        }
    }
}
